entities/src/main/java/com/flipkart/varadhi/auth/RoleBindingNode.java

```diff
@@ -0,0 +1,38 @@
+package com.flipkart.varadhi.auth;
+
+import com.flipkart.varadhi.ValidateVaradhiResource;
+import com.flipkart.varadhi.entities.ResourceType;
+import com.flipkart.varadhi.entities.VaradhiResource;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+@Getter
+@EqualsAndHashCode(callSuper = true)
+@ValidateVaradhiResource(message = "Invalid Role Binding name. Check naming constraints.")
+public class RoleBindingNode extends VaradhiResource {
+    String resourceId;
+    ResourceType resourceType;
+    Map<String, Set<String>> subjectToRolesMapping;
+
+    public RoleBindingNode(
+            String resourceId, ResourceType resourceType, Map<String, Set<String>> subjectToRolesMapping, int version
+    ) {
+        super(resourceId, version);
+        this.resourceId = resourceId;
+        this.resourceType = resourceType;
+        this.subjectToRolesMapping = new HashMap<>();
```

Comment:
need to ensure that `subjectsToRoleMapping` is modifiable map. It is possible to instantiate by passing unmodifiable map via `Map.of()` which will cause the `setRoleAssignment` method to throw unmodifiable collection exception